container/src/main/java/org/openremote/container/Container.java

```diff
@@ -283,7 +293,12 @@ public <T extends ContainerService> T getService(Class<T> type) throws IllegalSt
     }
 
     @Override
-    public ScheduledExecutorService getExecutorService() {
-        return EXECUTOR_SERVICE;
+    public ScheduledExecutorService getScheduledExecutor() {
+        return SCHEDULED_EXECUTOR;
+    }
+
+    @Override
+    public ExecutorService getExecutor() {
+        return EXECUTOR;
```

Comment:
Can you document what these executors are typically used for as you can submit runnables to either of them?
When you do not know the difference, you might end up just using the `SCHEDULED_EXECUTOR` for everything because the `ScheduledExecutorService` class extends `ExecutorService` so it can be used for everything.

Reply:
You would think they are the same for the reason you mentioned but they are different beasts, if you try using a `ScheduledExecutorService` for submitting runnables you will quickly discover they don't execute.

I haven't read too much into the logic for having both within java or why the scheduled implementation didn't behave the same for submitted runnables but I have seen both used in other frameworks as well.

Reply:
But does it matter as a developer which one you use or can you use whatever executor you want?

E.g. you can also submit a task with a delay of 0 to a `ScheduledExecutorService` to get it executed as fast as possible instead of using an `ExecutorService`.

The issue is that if you use start using the wrong executor, the allocated thread pool might not be big enough and you need to have an idea what tasks run on it to know what would be a reasonable size.

Having to deal with two thread pools may also create unneccesary threads which are sitting idle while the other thread pool is exhausted and could use more threads.

What usually makes sense is to have a different thread pools to make sure certain threads are guaranteed to run without being impacted by other threads.

Without any documentation I would think that these are just generic reusable thread pools and you can use them for whatever you want?

Does some (external?) code use an `ExecutorService` while other code uses an `ScheduledExecutorService` and did these pools come into existence to be able to share the threads in order to bring down the total number of threads?

Reply:
I've now added some javadoc for the executors, key point is the `SCHEDULED_EXECUTOR` is a `ScheduledThreadPoolExecutor` which is a fixed pool implementation in java and this is intended for scheduled task execution only whilst the `EXECUTOR` is a dynamic pool which is for general task execution with a standardised approach to thread management.